godot-git-plugin/src/git_api.cpp

```diff
@@ -203,39 +229,279 @@ Dictionary GitAPI::_get_modified_files_data() {
 		}
-		switch (entry->status) {
 
-			case GIT_STATUS_INDEX_NEW:
-			case GIT_STATUS_WT_NEW: {
+		static Dictionary map_changes;
+		map_changes[GIT_STATUS_WT_NEW] = 0;
+		map_changes[GIT_STATUS_INDEX_NEW] = 0;
+		map_changes[GIT_STATUS_WT_MODIFIED] = 1;
+		map_changes[GIT_STATUS_INDEX_MODIFIED] = 1;
+		map_changes[GIT_STATUS_WT_RENAMED] = 2;
+		map_changes[GIT_STATUS_INDEX_RENAMED] = 2;
+		map_changes[GIT_STATUS_WT_DELETED] = 3;
+		map_changes[GIT_STATUS_INDEX_DELETED] = 3;
+		map_changes[GIT_STATUS_WT_TYPECHANGE] = 4;
+		map_changes[GIT_STATUS_INDEX_TYPECHANGE] = 4;
+		map_changes[GIT_STATUS_CONFLICTED] = 5;
+
+		const static int GIT_STATUS_WT = GIT_STATUS_WT_NEW | GIT_STATUS_WT_MODIFIED | GIT_STATUS_WT_DELETED | GIT_STATUS_WT_TYPECHANGE | GIT_STATUS_WT_RENAMED | GIT_STATUS_CONFLICTED;
+		const static int GIT_STATUS_INDEX = GIT_STATUS_INDEX_NEW | GIT_STATUS_INDEX_MODIFIED | GIT_STATUS_INDEX_DELETED | GIT_STATUS_INDEX_RENAMED | GIT_STATUS_INDEX_TYPECHANGE;
+
+		if (entry->status & GIT_STATUS_WT) {
+			wt[path] = map_changes[entry->status & GIT_STATUS_WT];
+		}
 
-				diff[path] = 0;
-			} break;
-			case GIT_STATUS_INDEX_MODIFIED:
-			case GIT_STATUS_WT_MODIFIED: {
+		if (entry->status & GIT_STATUS_INDEX) {
+			if (entry->status == GIT_STATUS_INDEX_RENAMED) {
+				String old_path = entry->head_to_index->old_file.path;
+				index[old_path] = map_changes[GIT_STATUS_INDEX_DELETED];
+				index[path] = map_changes[GIT_STATUS_INDEX_NEW];
+				continue;
+			}
+			index[path] = map_changes[entry->status & GIT_STATUS_INDEX];
+		}
+	}
 
-				diff[path] = 1;
-			} break;
-			case GIT_STATUS_INDEX_RENAMED:
-			case GIT_STATUS_WT_RENAMED: {
+	diff["index"] = index;
+	diff["wt"] = wt;
+
+	git_status_list_free(statuses);
+	return diff;
+}
 
-				diff[path] = 2;
-			} break;
-			case GIT_STATUS_INDEX_DELETED:
-			case GIT_STATUS_WT_DELETED: {
+Array GitAPI::_get_branch_list() {
+	git_branch_iterator *it;
+	git_reference *ref;
+	git_branch_t type;
+	Array branch_names;
+	git_branch_iterator_new(&it, repo, GIT_BRANCH_LOCAL);
 
-				diff[path] = 3;
-			} break;
-			case GIT_STATUS_INDEX_TYPECHANGE:
-			case GIT_STATUS_WT_TYPECHANGE: {
+	branch_names.push_back(String()); // Leave Space for current branch
 
-				diff[path] = 4;
-			} break;
+	while (git_branch_next(&ref, &type, it) != GIT_ITEROVER) {
+		const char *name;
+		git_branch_name(&name, ref);
+		if (git_branch_is_head(ref)) {
+			branch_names[0] = String(name);
+		} else {
+			branch_names.push_back(String(name));
 		}
+		git_reference_free(ref);
 	}
+	git_branch_iterator_free(it);
 
-	git_status_list_free(statuses);
+	return branch_names;
+}
 
-	return diff;
+const char *GitAPI::_get_current_branch_name(bool full_ref) {
+	git_reference *head, *branch;
+	git_repository_head(&head, repo);
+	git_reference_resolve(&branch, head);
+	const char *branch_name;
+	if (full_ref) {
+		branch_name = git_reference_name(branch);
+	} else {
+		git_branch_name(&branch_name, branch);
+	}
+	return branch_name;
+}
```

Comment:
Is freeing the references not required here? I think freeing them has been done elsewhere too